### ISSUE TITLE
Restore is not working without a restart

### ISSUE THREAD

**Issue by cschockaert** [gentics/mesh#523]
## Gentics Mesh Version, operating system, or hardware.
- v0.21.5

## Operating System
- Linux
- MacOSX
- Windows
- Other, name?

## JVM
- Oracle JDK 1.8.0_100
- Open JDK 1.8.0_100

## Problem

After restore via mesh cli we have theses errors in log:

```
cms_mesh_master.1.m1e5ylihqpru@lc    | $ANSI{green {db=OSystem}} Database is open by another process
cms_mesh_master.1.m1e5ylihqpru@lc    | java.nio.channels.OverlappingFileLockException
cms_mesh_master.1.m1e5ylihqpru@lc    | 	at sun.nio.ch.SharedFileLockTable.checkList(FileLockTable.java:255)
cms_mesh_master.1.m1e5ylihqpru@lc    | 	at sun.nio.ch.SharedFileLockTable.add(FileLockTable.java:152)
cms_mesh_master.1.m1e5ylihqpru@lc    | 	at sun.nio.ch.FileChannelImpl.tryLock(FileChannelImpl.java:1108)
cms_mesh_master.1.m1e5ylihqpru@lc    | 	at java.nio.channels.FileChannel.tryLock(FileChannel.java:1155)
cms_mesh_master.1.m1e5ylihqpru@lc    | 	at com.orientechnologies.orient.core.storage.impl.local.paginated.OPaginatedStorageDirtyFlag.lockFile(OPaginatedStorageDirtyFlag.java:90)
cms_mesh_master.1.m1e5ylihqpru@lc    | 	at com.orientechnologies.orient.core.storage.impl.local.paginated.OPaginatedStorageDirtyFlag.open(OPaginatedStorageDirtyFlag.java:129)
cms_mesh_master.1.m1e5ylihqpru@lc    | 	at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.preOpenSteps(OLocalPaginatedStorage.java:424)
cms_mesh_master.1.m1e5ylihqpru@lc    | 	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.open(OAbstractPaginatedStorage.java:271)
cms_mesh_master.1.m1e5ylihqpru@lc    | 	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.open(ODatabaseDocumentTx.java:259)
cms_mesh_master.1.m1e5ylihqpru@lc    | 	at com.orientechnologies.orient.server.OServer.openDatabaseBypassingSecurity(OServer.java:1006)
cms_mesh_master.1.m1e5ylihqpru@lc    | 	at com.orientechnologies.orient.server.OServer.openDatabase(OServer.java:973)
cms_mesh_master.1.m1e5ylihqpru@lc    | 	at com.orientechnologies.orient.server.OServer.openDatabase(OServer.java:958)
cms_mesh_master.1.m1e5ylihqpru@lc    | 	at com.orientechnologies.orient.server.OSystemDatabase.openSystemDatabase(OSystemDatabase.java:92)
cms_mesh_master.1.m1e5ylihqp

```
## Expected behaviour and actual behaviour

I think a restart will correct the problem.

**Comment by Jotschi** [comment]
I assume mesh is running in a cluster env? Can you reproduce the issue with 0.27.0? I have just recently tested the backup and restore with 0.27.0 and did not encounter the issue.

**Comment by Jotschi** [comment]
Possible implementation options:
1. Shutdown the OrientDB server in the Mesh instance which is running the restore
2. Run the restore
3. Restart the OrientDB server

Questions:
* Will the db be dropped due to the quorum rule? Other instances overrule the restored instance. How can we deal with this situation. See https://github.com/orientechnologies/orientdb/issues/8631
* Should we alter the cluster group name to provide an easy way to migrate new instances to the new data set? (e.g. shut down "old" instances and run "new" instances which replicate data from the restored instance.)

Issue may be related to #510

**Comment by cschockaert** [comment]
> I assume mesh is running in a cluster env? Can you reproduce the issue with 0.27.0?

@Jotschi we were in standalone mode but restore occured when getmesh was heavy loaded (40 req / s average)

After a restart it returns working properly without lock

Only certains type of schema were locked, i created a new schema and new node and it worked fine.

Perhaps because not all orientdb files were locked ?

Another thing: during the restore, i saw some errors logs about orientdb was locked too.
mesh cli returned status 504 too.

But in the end restore was effective (we manually checked that modified nodes during the current day were restored to precedent version).

Hope that PR about restoring older node version will become effective soon ;)

**Comment by Jotschi** [comment]
@cschockaert I see. In the standalone setup. Did you have `storage.startServer` or the env `MESH_GRAPH_STARTSERVER` set to true?

Gentics Mesh can only handle read requests when a __backup__ is running. 

At the moment there are no checks if the __restore__ is invoked. I however think it would be best to not accept any requests during the time of the restore. I need to take a closer look at that feature to determine which mode results in the most reliable restore. If you need to restore at the moment I recommend to restore your db in a standalone instance which does not handle any requests. Once the restore has finishes you can switch on clustering and form a new cluster. (e.g. via changing the cluster name - otherwise the existing instances with the old data may overrule the restored instance).

**Comment by cschockaert** [comment]
@Jotschi 

yes MESH_GRAPH_STARTSERVER is true,

this is why we get theses errors?

**Comment by Jotschi** [comment]
@Jotschi I have not yet investigated the issue in detail but I assume that you won't see the error if you run single node instance with `MESH_GRAPH_STARTSERVER=false`. This is also the way the restore process is currently automatically tested. I have to add other tests in which the server is started to see where the issue is originating from.

**Comment by Jotschi** [comment]
I added a check which prevents restore operations in clustered mode. This change was released with 0.29.4
